lib/srv/alpnproxy/proxy.go

```diff
@@ -327,10 +328,13 @@ func (p *Proxy) Serve(ctx context.Context) error {
 				if cerr := clientConn.Close(); cerr != nil && !utils.IsOKNetworkError(cerr) {
 					p.log.WithError(cerr).Warnf("Failed to close client connection.")
 				}
-
-				if trace.IsBadParameter(err) {
+				switch {
+				case trace.IsBadParameter(err):
 					p.log.Warnf("Failed to handle client connection: %v", err)
```

Comment:
Is there a particular reason why we use '%v' rather than `.WithError` here ? Could we just remove this case and have it fall through to the default ?

Reply:
Yes, The `p.log.WithError(err).Warnf` will dump stack trace  where in case of IsBadParameter we want to just print the error message without duping stack trace.